File: Industrial-Programming/pr4/pr4_6.cpp
#include <regex>
#include <string>
#include <iostream>
#include <fstream>
#include <streambuf>

int main() {
    std::ifstream file("index.html");
    std::string html((std::istreambuf_iterator<char>(file)),
                      std::istreambuf_iterator<char>());

    std::regex imgRegex(R"|(<img[^>]+src\s*=\s*['"]([^'"]+)['"][^>]*>)|");
    std::sregex_iterator it(html.begin(), html.end(), imgRegex), end;

    for (; it != end; ++it) {
        std::cout << "Image URL: " << it->str(1) << std::endl;
    }

    return 0;
}

File: Industrial-Programming/pr4/pr4_6.js
 //6

const html = `
    <html>
        <body>
            <img src="Image1.jpg" alt="Image 1">
            <img src="Image2.jpg" alt="Image 2">
        </body>
    </html>
`;

const imgRegex = /<img[^>]+src\s*=\s*['"]([^'"]+)['"][^>]*>/g;
const matches = html.matchAll(imgRegex);

for (const match of matches) {
    console.log("Image URL: " + match[1]);
}
